compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt

```diff
@@ -124,15 +102,12 @@ class ClassCodegen private constructor(
         }
     }
 
-    private var hasAssertField = irClass.hasAssertionsDisabledField(context)
-    private var classInitializer = irClass.functions.singleOrNull { it.name.asString() == "<clinit>" }
-    private var generatingClInit = false
     private var generated = false
 
-    fun generate(parentDelegatedPropertyTracker: DelegatedPropertyOptimizer? = null): ReifiedTypeParametersUsages {
+    fun generate(parentDelegatedPropertyTracker: DelegatedPropertyOptimizer? = null) {
```

Comment:
FYI passing `DelegatedPropertyOptimizer` like this is incorrect - the description of #3629 is wrong, and inline functions cannot in fact be excluded:
```
import kotlin.reflect.KProperty

inline operator fun String.getValue(t:Any?, p: KProperty<*>): String = p.name + this

object ForceOutOfOrder {
    // This call forces `C$inlineFun$1` to be generated before `C` itself.
    // If `IrSourceCompilerForInline` passes a `null` optimizer to `generateMethodNode`,
    // the use of `C.$$delegatedProperties` will not be recorded anywhere.
    fun callInline() = C.inlineFun()
}

object C {
    inline fun inlineFun() = {
        val O by "K"
        O
    }()
}

fun box(): String = ForceOutOfOrder.callInline() // NoSuchFieldError
```
I will fix this in a separate PR.

Reply:
This is also problematic if `inlineFun` is called from a different package, as `$$delegatedProperties` has package-private visibility...

Reply:
Thanks! Reported the first problem as [KT-42253](https://youtrack.jetbrains.com/issue/KT-42253), the second one looks like [KT-39434](https://youtrack.jetbrains.com/issue/KT-39434).